src/kernels/MIOpenTensorKernels.cl

```diff
@@ -166,8 +170,12 @@ __kernel void OpTensorFwdBiasGeneric(global MIOPEN_TYPE* a,
             int o_w    = (incr_wg == 1) ? (lid % c_w) : ((lid / c_n) % c_w);
             int aindex = o_n * a_nstride + o_c * a_cstride + o_h * a_hstride + o_w;
             int cindex = o_n * c_nstride + o_c * c_cstride + o_h * c_hstride + o_w;
+#if MIO_BETA_ZERO == 1
```

Comment:
Conceptually, the `MIO_BETA_ZERO` macro is a part of the kernel's API. Currently the semantics of the macro is something like: "_If MIO_BETA_ZERO is defined as 1, then kernel ignores input values of tensor C (and thus the kernel does not produce NaNs even if C contains them). If MIO_BETA_ZERO is undefined or has any other value, then C is processed as usual._"

I recommend making this API more rigorous, requiring programmers to clearly understand and express what they want. This will reduce the chance of unintentional bugs: "_If MIO_BETA_ZERO is defined as 1, then kernel ignores input values of tensor C (and thus the kernel does not produce NaNs even if C contains them). If MIO_BETA_ZERO is defined as 0, then C is processed as usual._ ___If MIO_BETA_ZERO is undefined or has any other value, then kernel fails to build.___"

This can be expressed explicitly by adding the following near the beginning of the file:
```C++
#ifndef MIO_BETA_ZERO
#error "MIO_BETA_ZERO must be defined"
#endif
#if MIO_BETA_ZERO != 1 || MIO_BETA_ZERO != 0
#error "MIO_BETA_ZERO value is invalid"
#endif
```

The above opens the way for
```suggestion
#if MIO_BETA_ZERO
```
instead `#if MIO_BETA_ZERO == 1`. We use this approach with macros that carry boolean semantics in all modern kernels (and instead of `#if MACRO == 0` we normally use `#if !MACRO`). **_And I highly recommend using it here_**.

Note that if you still would like to stay with the current semantics of MIO_BETA_ZERO, then the following code can be used to implement the necessary checks and normalization:
```C++
#if defined(MIO_BETA_ZERO) && MIO_BETA_ZERO != 1
#undef MIO_BETA_ZERO
#endif
#ifndef MIO_BETA_ZERO
#define MIO_BETA_ZERO 0
#endif
```

Reply:
yes, I will add below code with the current semantics of MIO_BETA_ZERO.

#if defined(MIO_BETA_ZERO) && MIO_BETA_ZERO != 1
#undef MIO_BETA_ZERO
#endif
#ifndef MIO_BETA_ZERO
#define MIO_BETA_ZERO 0
#endif